### PR TITLE
30 admin merchants top 5 merchants by revenue

### PR DESCRIPTION
**Pull request by Jesse193**
This adds the top 5 merchants to the admin merchant index page

### CODE REVIEW

**Review by amspears007 — APPROVED** [review]
Yay!   it was a transactions/invoices join, great job ya'll

File: app/models/merchant.rb [hunk 1]
```diff
@@ -36,4 +36,8 @@ def current_status
       "Enabled"
     end
   end
+
+  def self.top_five_merchants
+    joins(invoices: :transactions).where(transactions: { result: "success"}).select("merchants.id, merchants.name, SUM(invoice_items.unit_price * invoice_items.quantity) as total").group(:id).order("total desc").limit(5)
```
Comment (kaileykaes) [inline app/models/merchant.rb:41]: Huh. Researched this a bit. Interesting that sum calculates products in postgresql

File: app/views/admin/merchants/_form.html.erb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<%= form_with url: "/admin/merchants/#{@merchant.id}", method: method do |f| %>
```
Comment (kaileykaes) [inline app/views/admin/merchants/_form.html.erb:1]: nice refactor

File: app/views/admin/merchants/new.html.erb [hunk 3]
```diff
@@ -1,6 +1 @@
-<%= form_with url: "/admin/merchants", method: :post do |f| %>
-  <%= f.label :name %>
-  <%= f.text_field :name %>
-
-  <%= f.submit "Submit" %>
-<% end %>
+<%= render partial: 'form', locals: {url: "/admin/merchants", model: @merchant, method: :post} %>
```
Comment (kaileykaes) [inline app/views/admin/merchants/new.html.erb:1]: neat implementation of your refactored partial!

File: config/routes.rb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 Rails.application.routes.draw do
   
-  get "/admin", to: "admin/dashboards#dashboard" 
+  get "/admin", to: "admin/dashboards#index"
```
Comment (kaileykaes) [inline config/routes.rb:3]: RESTful. Approve.

File: db/data/invoice_items.csv [hunk 5]
```diff
@@ -891,8 +891,7 @@ id,item_id,invoice_id,quantity,unit_price,status,created_at,updated_at
 890,925,198,2,38716,packaged,2012-03-27 14:54:19 UTC,2012-03-27 14:54:19 UTC
 891,938,198,5,57632,packaged,2012-03-27 14:54:19 UTC,2012-03-27 14:54:19 UTC
 892,928,198,9,32346,packaged,2012-03-27 14:54:19 UTC,2012-03-27 14:54:19 UTC
-893,927,198,8,96437,shipped,2012-03-27 14:54:19 UTC,2012-03-27 14:54:19 UTC
```
Comment (kaileykaes) [inline db/data/invoice_items.csv:894]: Did we intend to lose 893?

**Review by kaileykaes — APPROVED** [review]

**Pull request merged**